msg_helpers.go

```diff
@@ -53,9 +44,7 @@ func unpackDataAAAA(msg []byte, off int) (net.IP, int, error) {
 	if off+net.IPv6len > len(msg) {
 		return nil, len(msg), &Error{err: "overflow unpacking aaaa"}
 	}
-	aaaa := net.IP{msg[off], msg[off+1], msg[off+2], msg[off+3], msg[off+4],
-		msg[off+5], msg[off+6], msg[off+7], msg[off+8], msg[off+9], msg[off+10],
-		msg[off+11], msg[off+12], msg[off+13], msg[off+14], msg[off+15]}
+	aaaa := net.IP(msg[off : off+net.IPv6len])
```

Comment:
Even after adding the append(....) you suggested I'm seeing better performance vs calling net.IPv4(...).